### DISCUSSION TITLE
How to configure this to make it work with Swift?

### DISCUSSION THREAD

**Discussion by konrad1977** [emacs-tree-sitter/elisp-tree-sitter#207] [Q&A] [answered]
I cannot get it to work with swift. It hooks in but nothing changes.

If I check my *message* I can see message like this. 

> [tree-sitter-langs] Running (git rev-parse --show-toplevel) in /Users/mikaelkonradsson/.emacs.d/elpa/tree-sitter-langs-20211228.1446/
> The directory /Users/mikaelkonradsson/.emacs.d/elpa/tree-sitter-langs-20211228.1446/ doesn’t seem to be a git working dir. Grammar-building functions will not work.
> tree-sitter-langs: Grammar bundle v0.10.13 was already installed; skipped

And if I check the tree-sitter-tree:
<img width="1792" alt="Skärmavbild 2022-01-29 kl  09 46 42" src="https://user-images.githubusercontent.com/4384932/151660149-3c3675eb-1b59-480d-b201-171987bda038.png">

My config block for tree-sitter:

```
(use-package tree-sitter
  :config
  (add-to-list 'tree-sitter-major-mode-language-alist '(swift-mode . swift))
  (add-hook 'swift-mode-hook #'tree-sitter-mode)
  (add-hook 'tree-sitter-after-on-hook #'tree-sitter-hl-mode))

(use-package tree-sitter-langs
  :after (tree-sitter))


```

**Comment by ubolonton** [comment, marked as answer]
The screenshot shows the syntax tree to the right. That means it's working.

If you mean code highlighting didn't change, that's because it should be provided by another package.
`tree-sitter-langs` doesn't provide highlighting [queries](https://github.com/emacs-tree-sitter/tree-sitter-langs/tree/master/queries) for Swift.

**Reply by konrad1977** [reply]
Maybe we/you should add queries (ask the author of the current swift-package) or change to the one I suggested which has queries but for some reason emacs doesn't change the foreground color, only background color. (dunno why though). The latter package is really slow on my intel Mac though. Haven't tried on my M1Max. But it takes around 10 second to view a file.
